domain/aggregator/scan_state_aggregator.go

```diff
@@ -0,0 +1,244 @@
+/*
+ * © 2025 Snyk Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package aggregator
+
+import (
+	"sync"
+
+	"github.com/snyk/snyk-ls/application/config"
+	"github.com/snyk/snyk-ls/internal/product"
+)
+
+// FolderProductKey identifies a unique (FolderPath, ProductName) pair.
+type FolderProductKey struct {
+	FolderPath string
+	Product    product.Product
+}
+
+type ScanStatus string
+
+const (
+	NotStarted ScanStatus = "NOT_STARTED"
+	InProgress ScanStatus = "IN_PROGRESS"
+	Success    ScanStatus = "Success"
+	Error      ScanStatus = "ERROR"
+)
+
+// ScanState describes the state for one scan (per folder+product).
+type ScanState struct {
+	Status ScanStatus
+	Err    error
+}
+
+type ScanStateMap map[FolderProductKey]*ScanState
+
+// ScanStateAggregator stores and manages the scan states for working directory and reference scans.
+type ScanStateAggregator struct {
+	mu                         sync.RWMutex
+	referenceScanStates        ScanStateMap
+	workingDirectoryScanStates ScanStateMap
+	scanStateChangeEmitter     ScanStateChangeEmitter
+	c                          *config.Config
+}
+
+// NewScanStateAggregator constructs a new aggregator.
+func NewScanStateAggregator(ssce ScanStateChangeEmitter, c *config.Config) StateAggregator {
```

Comment:
I'm nitpicking, I know, but could we have `c *config.Config` always as first param?